dashboard/app/models/lti_user_identity.rb

```diff
@@ -24,4 +24,5 @@ class LtiUserIdentity < ApplicationRecord
   has_and_belongs_to_many :lti_deployments
 
   validates :subject, presence: true
+  validates :subject, uniqueness: {scope: :lti_integration_id}
```

Comment:
Uniqueness validation triggers a DB request on each save to check if the record is unique.
It makes sense to check this only when the relevant fields have changed for optimization.
```suggestion
  validates :subject, uniqueness: {scope: :lti_integration_id}, if: -> {lti_integration_id_changed? || subject_changed?}
```